src/System.Linq.Expressions/tests/BinaryOperators/Assignment/Assign.cs

```diff
@@ -24,20 +25,77 @@ private class PropertyAndFields
             public string WriteOnlyString { set { } }
             public static string WriteOnlyStaticString { set { } }
 
+            public BaseClass BaseClassProperty { get; set; }
+
             public int Int32Property { get; set; }
             public int Int32Field;
             public readonly int ReadonlyInt32Field;
             public int ReadonlyInt32Property { get { return 42; } }
-            public static int StaticInt32Property { get; set; }
+            public static int StaticInt32Property1 { get; set; }
             public static int StaticInt32Field;
             public static readonly int ReadonlyStaticInt32Field;
             public static int ReadonlyStaticInt32Property { get { return 321; } }
             public const int ConstantInt32 = 12;
             public int WriteOnlyInt32 { set { } }
             public static int WriteOnlyStaticInt32 { set { } }
+
+            private int _underlyingIndexerField1;
+            public int this[int i]
+            {
+                get { return _underlyingIndexerField1; }
+                set
+                {
+                    if (i == 1)
+                    {
+                        _underlyingIndexerField1 = value;
+                    }
+                }
```

Comment:
Should this (and the other indexers you added) throw/assert for any other inputs?